include/envoy/api/api.h

```diff
@@ -47,6 +47,11 @@ class Api {
    * @return a reference to the TimeSource
    */
   virtual TimeSource& timeSource() PURE;
+
+  /**
+   * @return a reference to the Stats::Store
+   */
+  virtual Stats::Store& statsStore() PURE;
```

Comment:
Q: Why do we need/want stats store as part of the API? This is a little different from the other API objects in that it's a constructed singleton? Mostly just looking for clarification for my own understanding.

Reply:
Well, we already have the `FactoryContext` etc. which we pass to filters for this purpose, and I'm not sure we want to pass the entire `Store` to most extensions, vs. a `Scope`, etc. I think my feeling is probably to take this out and have extensions obtain these type of potentially scoped singletons from the factory context they are passed? WDYT?

Reply:
OK, sgtm. I was thinking we wanted to funnel everything through API at some point but that's kind of speculative. Removed.